src/fides/api/api/v1/endpoints/connection_endpoints.py

```diff
@@ -217,19 +239,47 @@ def put_connection_config_secrets(
     """
     connection_config = get_connection_config_or_error(db, connection_key)
 
-    connection_config.secrets = validate_secrets(
-        db, unvalidated_secrets, connection_config
-    ).dict()
-    # Save validated secrets, regardless of whether they've been verified.
-    logger.info("Updating connection config secrets for '{}'", connection_key)
-    connection_config.save(db=db)
+    return validate_and_update_secrets(
+        connection_key, connection_config, db, unvalidated_secrets, verify
+    )
 
-    msg = f"Secrets updated for ConnectionConfig with key: {connection_key}."
 
-    if verify:
-        return connection_status(connection_config, msg, db)
+@router.patch(
+    CONNECTION_SECRETS,
+    status_code=HTTP_200_OK,
+    dependencies=[Security(verify_oauth_client, scopes=[CONNECTION_CREATE_OR_UPDATE])],
+    response_model=TestStatusMessage,
+)
+def patch_connection_config_secrets(
+    connection_key: FidesKey,
+    *,
+    db: Session = Depends(deps.get_db),
+    unvalidated_secrets: connection_secrets_schemas,
+    verify: Optional[bool] = True,
+) -> TestStatusMessage:
+    """
+    Partially update secrets that will be used to connect to a specified connection_type.
 
-    return TestStatusMessage(msg=msg, test_status=None)
+    The specific secrets will be connection-dependent. For example, the components needed to connect to a Postgres DB
+    will differ from Dynamo DB.
+    """
+    connection_config = get_connection_config_or_error(db, connection_key)
+
+    existing_secrets = connection_config.secrets
+
+    # We create the new secrets object by combining the existing secrets with the new secrets.
+    patched_secrets = {}
+    if existing_secrets:
+        patched_secrets = {**existing_secrets}  # type: ignore[arg-type]
```

Comment:
For some reason, the type checker shows that `existing_secrets` has a type of `Optional[Any]` ... not sure why since the column is defined as a Dict column. Any help on how to fix this -- without just ignoring the arg type, which is what I'm doing here -- is appreciated 😄

Reply:
You might be able to fix it with just 
```
 existing_secrets: Dict[str, str] = connection_config.secrets
```

Reply:
that yields this error
```
src/fides/api/api/v1/endpoints/connection_endpoints.py:268: error: Incompatible types in assignment (expression has type "Optional[Any]", variable has type "Dict[str, Any]")  [assignment]
        existing_secrets: Dict[str, Any] = connection_config.secrets
                                           ^~~~~~~~~~~~~~~~~~~~~~~~~
```

Reply:
solved in https://github.com/ethyca/fides/pull/5172/commits/90211f195c957156b4bd38e682af494c22b129ef